### PR TITLE
Controller manager args for clusterapi

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*
#488 
*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clusterapi/controllermanagerargs_test.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+						},
+					},
+				},
+				DatacenterRef: v1alpha1.Ref{
+					Kind: "SnowDatacenterConfig",
+					Name: "test",
+				},
+			},
+		}
+		s.SnowDatacenter = givenDatacenterConfig()
```
Comment (jiayiwang7) [inline pkg/clusterapi/controllermanagerargs_test.go:103]: do we need `s.SnowDatacenter`, `s.SnowMachineConfigs`, `s.VersionsBundle` and `s.ManagementCluster` for this test?

Comment (mitalipaygude) [reply pkg/clusterapi/controllermanagerargs_test.go]: not really, removing

**Review by jiayiwang7 — APPROVED** [review]

**Review by jiayiwang7 — APPROVED** [review]

File: pkg/clusterapi/apibuilder.go [hunk 2]
```diff
@@ -140,7 +140,7 @@ func KubeadmControlPlane(clusterSpec *cluster.Spec, infrastructureObject APIObje
 						},
 					},
 					ControllerManager: bootstrapv1.ControlPlaneComponent{
-						ExtraArgs: map[string]string{},
+						ExtraArgs: SetControllerManagerArgs(clusterSpec),
```
Comment (jiayiwang7) [inline pkg/clusterapi/apibuilder.go:143]: don't you need it for kubelet extra arg as well?

File: pkg/clusterapi/apibuilder_test.go [hunk 3]
```diff
@@ -225,7 +225,7 @@ func wantKubeadmControlPlane() *controlplanev1.KubeadmControlPlane {
 						},
 					},
 					ControllerManager: bootstrapv1.ControlPlaneComponent{
-						ExtraArgs: map[string]string{},
+						ExtraArgs: clusterapi.SecureTlsCipherSuitesExtraArgs(),
```
Comment (g-gaston) [inline pkg/clusterapi/apibuilder_test.go:228]: can we have here hardcoded the extra args in a map?

File: pkg/clusterapi/controllermanagerargs.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"github.com/aws/eks-anywhere/pkg/cluster"
+)
+
+func SetControllerManagerArgs(clusterSpec *cluster.Spec) ExtraArgs {
+	controllerManagerExtraArgs := SecureTlsCipherSuitesExtraArgs().
+		Append(NodeCIDRMaskExtraArgs(&clusterSpec.Cluster.Spec.ClusterNetwork))
+
+	return controllerManagerExtraArgs
```
Comment (g-gaston) [inline pkg/clusterapi/controllermanagerargs.go:11]: nit: 
```suggestion
	return SecureTlsCipherSuitesExtraArgs().
		Append(NodeCIDRMaskExtraArgs(&clusterSpec.Cluster.Spec.ClusterNetwork))
```

File: pkg/clusterapi/identity_test.go [hunk 5]
```diff
@@ -105,7 +105,7 @@ func TestConfigureAWSIAMAuthInKubeadmControlPlane(t *testing.T) {
 								},
 							},
 							ControllerManager: bootstrapv1.ControlPlaneComponent{
-								ExtraArgs: map[string]string{},
+								ExtraArgs: clusterapi.SecureTlsCipherSuitesExtraArgs(),
```
Comment (g-gaston) [inline pkg/clusterapi/identity_test.go:108]: same here about not using `clusterapi.SecureTlsCipherSuitesExtraArgs()`
we can have a method in the tests like `defaultControllerManagerTests()` so you can reuse that in all these tests if you want

File: pkg/clusterapi/controllermanagerargs.go [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package clusterapi
+
+import (
+	"github.com/aws/eks-anywhere/pkg/cluster"
+)
+
+func SetControllerManagerArgs(clusterSpec *cluster.Spec) ExtraArgs {
```
Comment (vivek-koppuru) [inline pkg/clusterapi/controllermanagerargs.go:7]: Why do we need a separate function and file for this? Do we anticipate this function to be exported by other packages in the future? I would argue that this is mainly a helper function in `apibuilder.go`

Comment (mitalipaygude) [reply pkg/clusterapi/controllermanagerargs.go:7]: These will eventually be used for all other providers as well.

File: pkg/clusterapi/apibuilder_test.go [hunk 7]
```diff
@@ -225,7 +225,7 @@ func wantKubeadmControlPlane() *controlplanev1.KubeadmControlPlane {
 						},
 					},
 					ControllerManager: bootstrapv1.ControlPlaneComponent{
-						ExtraArgs: map[string]string{},
+						ExtraArgs: DefaultControllerManagerArgs(),
```
[the remark this answers is not in the document]
Comment (mitalipaygude) [reply pkg/clusterapi/apibuilder_test.go:228]: using it in identity_test.go as well.

[the remark this answers is not in the document]
Comment (g-gaston) [reply pkg/clusterapi/apibuilder_test.go:228]: that's in the same package
Are you getting any compiler error?

**Review by jiayiwang7 — APPROVED** [review]

**Review by g-gaston — APPROVED** [review]

**Pull request merged**